src/network/networkd-ndisc.c

```diff
@@ -1176,30 +1176,37 @@ int config_parse_address_generation_type(
         if (r < 0)
                 return log_oom();
 
-        if ((p = startswith(rvalue, "static:")))
-                token->address_generation_type = IPV6_TOKEN_ADDRESS_GENERATION_STATIC;
-        else if ((p = startswith(rvalue, "prefixstable:")))
+        if ((p = startswith(rvalue, "prefixstable"))) {
                 token->address_generation_type = IPV6_TOKEN_ADDRESS_GENERATION_PREFIXSTABLE;
-        else {
+                if (*p == ':')
+                        p++;
+                else
```

Comment:
Hmm, so if I specify "IPv6Token=prefixstableasdfasdfasdf" then it will be interpreted as "prefixstable". I think this needs to check that `!*p` here.

Reply:
This does not a separate. It can be added there itself. Hence it is reductant